src/libcore/hash/mod.rs

```diff
@@ -664,13 +664,39 @@ mod impls {
         }
     }
 
+    #[unstable(feature = "unsized_pointer_hash", issue = "0")] // TODO
+    default impl<T: ?Sized> Hash for *const T {
+        fn hash<H: Hasher>(&self, state: &mut H) {
+            // We know T is unsized, because Sized types are handled
+            // by the specialized impl below.
+            let (a, b) = unsafe {
+                *(*self as *const T as *const (usize, usize))
```

Comment:
Doesn't this end up interpreting `T` itself as the `usize` pair?  i.e. `self` is `&*const T`, then `*self as *const T` is a redundant cast, then the final cast changes to the fat pointer pieces.

I think you want `*(self as *const *const T as *const (usize, usize))`

Reply:
Oops, you are right, of course.  Fixed this and added a test.